### PR TITLE
[DNM] *: add column support.

### PR DESCRIPTION
**Pull request by disksing**

### CODE REVIEW

File: src/storage/mvcc/mod.rs [hunk 1]
```diff
@@ -54,3 +54,41 @@ pub type Result<T> = ::std::result::Result<T, Error>;
 
 // Make sure meta version in tests could never catch up with key version(timestamp).
 pub const TEST_TS_BASE: u64 = 1000000;
+
+use kvproto::kvpb::{Row, RowValue, Mutation, Op};
+
+pub fn default_row(row_key: &[u8]) -> Row {
+    let mut row = Row::new();
+    row.set_row_key(row_key.to_owned());
+    row.mut_columns().push(vec![]);
```
Comment (siddontang) [inline src/storage/mvcc/mod.rs]: need push empty vector here?

File: src/storage/mvcc/mod.rs [hunk 2]
```diff
@@ -54,3 +54,41 @@ pub type Result<T> = ::std::result::Result<T, Error>;
...
+    row
+}
+
+pub fn row_value_empty(row_value: &RowValue) -> bool {
```
Comment (siddontang) [inline src/storage/mvcc/mod.rs]: is_row_value_empty

**Comment by ngaut** [comment]
PTAL @BusyJay

File: src/storage/mvcc/txn.rs [hunk 3]
```diff
@@ -12,18 +12,99 @@
...
+                            if let Some(x) = try!(self.read(&value_key)) {
+                                // TODO: Should we keep the order of columns in RowValue same as
+                                // Row?
+                                row_value.mut_columns().push(col.get_name().to_vec());
+                                row_value.mut_values().push(x);
+                            }
+                        }
+                    }
+                }
+            }
```
Comment (coocood) [inline src/storage/mvcc/txn.rs]: This block of code is an duplication of the code on line 65, can we unify them in the loop?
like 
```rust
let mut meta = first_meta;
loop {
    if let Some(x) = meta.iter_items().find(|x| x.get_commit_ts() <= ts) {
        ....
    }
    if pending_cols.len() == 0 {
         break;
    }
    next = meta.next_index();
    meta = match next {
        Some(x) => try!(self.load_meta(&row_key, x)),
        None => break,
    };
}

File: src/storage/mvcc/txn.rs [hunk 4]
```diff
@@ -12,18 +12,99 @@
...
+        }
+        let mut row_value = RowValue::new();
+        row_value.set_row_key(row.take_row_key());
+        let mut pending_cols = BTreeMap::new();
+        for col in row.get_columns() {
+            pending_cols.insert(col.to_vec(), true);
+        }
+        // Find the latest write below our start timestamp.
+        if let Some(x) = first_meta.iter_items().find(|x| x.get_commit_ts() <= ts) {
```
Comment (zhangjinpeng1987) [inline src/storage/mvcc/txn.rs]: find() returns the first element satisfying the specified predicate, some cols may get wrong version data or get no data

**Comment by disksing** [comment]
We are not going to merge the feature in Beta.

**Pull request closed without merging**